### PR TITLE
add support for discovery chain get method (#355)

### PR DESCRIPTION
**Pull request by larrytamnjong**

### CODE REVIEW

**Comment by larrytamnjong** [comment]
@marcin-krystianc

**Comment by marcin-krystianc** [comment]
Hi @larrytamnjong ,

please have a look at test errors, e.g.:
```
[xUnit.net 00:00:25.15]         --- End of stack trace from previous location ---
  Passed Consul.Test.CoordinateTest.Coordinate_GetNode [3 ms]
Error: Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get: Assert.Equal() Failure
  Failed Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get [21 ms]
  Error Message:
   Assert.Equal() Failure
          ↓ (pos 0)
Expected: https
Actual:   tcp
          ↑ (pos 0)
  Stack Trace:
     at Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get() in D:\a\consuldotnet\consuldotnet\Consul.Test\DiscoveryChainTest.cs:line 83
--- End of stack trace from previous location ---
```

**Comment by larrytamnjong** [comment]
> Hi @larrytamnjong ,
> 
> please have a look at test errors, e.g.:
> 
> ```
> [xUnit.net 00:00:25.15]         --- End of stack trace from previous location ---
>   Passed Consul.Test.CoordinateTest.Coordinate_GetNode [3 ms]
> Error: Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get: Assert.Equal() Failure
>   Failed Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get [21 ms]
>   Error Message:
>    Assert.Equal() Failure
>           ↓ (pos 0)
> Expected: https
> Actual:   tcp
>           ↑ (pos 0)
>   Stack Trace:
>      at Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get() in D:\a\consuldotnet\consuldotnet\Consul.Test\DiscoveryChainTest.cs:line 83
> --- End of stack trace from previous location ---
> ```

Hi @marcin-krystianc when I debug the test locally, the protocol in the response is  "https". However, I don't understand why it's "tcp" here. Could you provide some guidance on how to troubleshoot this issue?
image

**Comment by larrytamnjong** [comment]
> > Hi @larrytamnjong ,
> > please have a look at test errors, e.g.:
> > ```
> > [xUnit.net 00:00:25.15]         --- End of stack trace from previous location ---
> >   Passed Consul.Test.CoordinateTest.Coordinate_GetNode [3 ms]
> > Error: Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get: Assert.Equal() Failure
> >   Failed Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get [21 ms]
> >   Error Message:
> >    Assert.Equal() Failure
> >           ↓ (pos 0)
> > Expected: https
> > Actual:   tcp
> >           ↑ (pos 0)
> >   Stack Trace:
> >      at Consul.Test.DiscoveryChainTest.DiscoveryChaing_Get() in D:\a\consuldotnet\consuldotnet\Consul.Test\DiscoveryChainTest.cs:line 83
> > --- End of stack trace from previous location ---
> > ```
> 
> Hi @marcin-krystianc when I debug the test locally, the protocol in the response is "https". However, I don't understand why it's "tcp" here. Could you provide some guidance on how to troubleshoot this issue? image

@marcin-krystianc , I think I’ve found the issue that was causing the protocol to be "https" locally instead of "tcp." I've made an update to the test, and it should be resolved now. You can please take another look when you have time.

File: Consul/Interfaces/IDiscoveryChainEndpoint.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading.Tasks;
+
+namespace Consul.Interfaces
+{
+    /// <summary>
+    /// The interface for the Discovery Chain API Endpoints
+    /// </summary>
+    public interface IDiscoveryChainEndpoint
+    {
+        Task<QueryResult<DiscoveryChainResponse>> Get(string name, QueryOptions q, CancellationToken ct = default);
```
Comment (marcin-krystianc) [inline Consul/Interfaces/IDiscoveryChainEndpoint.cs:32]: We also need overrides for [compile-dc](https://developer.hashicorp.com/consul/api-docs/discovery-chain#compile-dc) 
 and other parameters (https://developer.hashicorp.com/consul/api-docs/discovery-chain#json-request-body-schema), but that ideally can be implemented in follow-up PR (so we limit the scope of this PR).

Comment (larrytamnjong) [reply Consul/Interfaces/IDiscoveryChainEndpoint.cs:32]: Okay got it 👍

**Pull request merged**